httpe/err.go

```diff
@@ -0,0 +1,142 @@
+	ErrPreconditionFailed           = errors.New(http.StatusText(http.StatusPreconditionFailed))
+	ErrRequestEntityTooLarge        = errors.New(http.StatusText(http.StatusRequestEntityTooLarge))
+	ErrRequestURITooLong            = errors.New(http.StatusText(http.StatusRequestURITooLong))
+	ErrUnsupportedMediaType         = errors.New(http.StatusText(http.StatusUnsupportedMediaType))
+	ErrRequestedRangeNotSatisfiable = errors.New(http.StatusText(http.StatusRequestedRangeNotSatisfiable))
+	ErrExpectationFailed            = errors.New(http.StatusText(http.StatusExpectationFailed))
+	ErrTeapot                       = errors.New(http.StatusText(http.StatusTeapot))
+	ErrMisdirectedRequest           = errors.New(http.StatusText(http.StatusMisdirectedRequest))
+	ErrUnprocessableEntity          = errors.New(http.StatusText(http.StatusUnprocessableEntity))
+	ErrLocked                       = errors.New(http.StatusText(http.StatusLocked))
+	ErrFailedDependency             = errors.New(http.StatusText(http.StatusFailedDependency))
+	ErrTooEarly                     = errors.New(http.StatusText(http.StatusTooEarly))
+	ErrUpgradeRequired              = errors.New(http.StatusText(http.StatusUpgradeRequired))
+	ErrPreconditionRequired         = errors.New(http.StatusText(http.StatusPreconditionRequired))
+	ErrTooManyRequests              = errors.New(http.StatusText(http.StatusTooManyRequests))
+	ErrRequestHeaderFieldsTooLarge  = errors.New(http.StatusText(http.StatusRequestHeaderFieldsTooLarge))
+	ErrUnavailableForLegalReasons   = errors.New(http.StatusText(http.StatusUnavailableForLegalReasons))
+
+	ErrInternalServerError           = errors.New(http.StatusText(http.StatusInternalServerError))
+	ErrNotImplemented                = errors.New(http.StatusText(http.StatusNotImplemented))
+	ErrBadGateway                    = errors.New(http.StatusText(http.StatusBadGateway))
+	ErrServiceUnavailable            = errors.New(http.StatusText(http.StatusServiceUnavailable))
+	ErrGatewayTimeout                = errors.New(http.StatusText(http.StatusGatewayTimeout))
+	ErrHTTPVersionNotSupported       = errors.New(http.StatusText(http.StatusHTTPVersionNotSupported))
+	ErrVariantAlsoNegotiates         = errors.New(http.StatusText(http.StatusVariantAlsoNegotiates))
+	ErrInsufficientStorage           = errors.New(http.StatusText(http.StatusInsufficientStorage))
+	ErrLoopDetected                  = errors.New(http.StatusText(http.StatusLoopDetected))
+	ErrNotExtended                   = errors.New(http.StatusText(http.StatusNotExtended))
+	ErrNetworkAuthenticationRequired = errors.New(http.StatusText(http.StatusNetworkAuthenticationRequired))
+)
+
+// WriteHTTPeErr writes the http status represented by err to the ResponseWriter and
+// prints the error message for 4xx errors, but keeps details for 5xx errors.
+func WriteHTTPeErr(w http.ResponseWriter, err error) { //nolint: funlen, gocyclo
+	switch {
+	case errors.Is(err, ErrBadRequest):
```

Comment:
this function and the error var initialisations could be be a lot simpler if you created your own error type for it:
```
type statusError int
func (err statusError) Error() string { return http.StatusText(err) }
func (err statusError) IsInternal() bool { return int(err) >= 500 && int(err) <= 599 }
func (err statusError) Code() int { return int(statusError) }
var (
  ErrBadRequest = statusError(http.StatusBadRequest)
  ...
)
```

Then you could write this function in terms of this type (with `errors.As()`) and remove all the error enumeration in the switch.

(and maybe statusError should be StatusError)

Reply:
ok that sounds good.
I thought about vaguely, but I thought it would force package users to use errors.As as well and that never flows as easily as errors.Is (two liner min). But what you suggest is kind of a mix of both. I like it.

Reply:
done.